Dependencies/xr/Source/OpenXR/XR.cpp

```diff
@@ -387,9 +387,76 @@ namespace xr
             renderResource.DepthSwapchains.clear();
         }
 
+        Anchor CreateAnchor(Pose pose, NativeTrackablePtr trackable, XrTime time)
+        {
+            auto& apiExtensions = HmdImpl.Extensions;
+
+            if (!apiExtensions->SpatialAnchorSupported)
+                throw std::runtime_error("Spatial anchors are not supported for this device.");
+
+            if (trackable != nullptr)
+                throw std::runtime_error("Anchors created from hit test results not yet implemented for OpenXR.");
+
+            auto anchorSpace = std::make_shared<OpenXRAnchorSpace>();
+
+            XrSpatialAnchorCreateInfoMSFT anchorCreateInfo{XR_TYPE_SPATIAL_ANCHOR_CREATE_INFO_MSFT};
+            anchorCreateInfo.space = SceneSpace;
+            anchorCreateInfo.pose = { pose.Orientation.X, pose.Orientation.Y, pose.Orientation.Z, pose.Orientation.W,
+                                      pose.Position.X, pose.Position.Y, pose.Position.Z };
+            anchorCreateInfo.time = time;
+
+            CHECK_XRCMD(apiExtensions->xrCreateSpatialAnchorMSFT(
+                Session,
+                &anchorCreateInfo,
+                anchorSpace->Anchor.Put(apiExtensions->xrDestroySpatialAnchorMSFT)));
+
+            XrSpatialAnchorSpaceCreateInfoMSFT anchorSpaceCreateInfo{XR_TYPE_SPATIAL_ANCHOR_SPACE_CREATE_INFO_MSFT};
+            anchorSpaceCreateInfo.anchor = anchorSpace->Anchor.Get();
+            anchorSpaceCreateInfo.poseInAnchorSpace = xr::math::Pose::Identity();
+            CHECK_XRCMD(apiExtensions->xrCreateSpatialAnchorSpaceMSFT(Session, &anchorSpaceCreateInfo, anchorSpace->Space.Put()));
+
+            const auto nativeAnchorPtr = reinterpret_cast<NativeAnchorPtr>(anchorSpace.get());
+            openXRAnchors[nativeAnchorPtr] = anchorSpace;
+            return { pose, nativeAnchorPtr };
+        }
+
+        void UpdateAnchor(Anchor& anchor, XrTime time) const
+        {
+            auto anchorSpace = openXRAnchors.at(anchor.NativeAnchor);
+
```

Comment:
Check if this key exists

Reply:
`.at` will throw an exception if the key isn't present in the map. Would we want to throw a different exception, perhaps with a string that explains the issue? Or would we want to do something like `if (itr == openXRAnchors.end()) return;` and skip the function entirely?